packages/common/src/events/_internal/shareEventWithGroups.ts

```diff
@@ -0,0 +1,77 @@
+import { IGroup, ISearchResult, searchGroups } from "@esri/arcgis-rest-portal";
+import { IArcGISContext } from "../../ArcGISContext";
+import { IHubEvent } from "../../core/types/IHubEvent";
+import { unique } from "../../util";
+import { updateEvent } from "../api/events";
+import { poll } from "../../utils/poll";
+
+/**
+ * Shares an event with one or more groups
+ * @param groupIds An array of group IDs to share the event with
+ * @param entity An IHubEvent object
+ * @param context An IArcGISContext object
+ * @returns a promise that resolves IHubItemEntity
+ */
+export async function shareEventWithGroups(
```

Comment:
consider adding more comments about what's going on in this - I'm not following why we're polling a search for groups etc, so my guess is that others on the team, including "Future Randy" will also be a little O_o when they need to get into this code :)

Reply:
Will do. For context, _new_ groups aren't _immediately_ available in the AGO search index. This prevents the need for us to add polling potentially in many places of the app where we create new groups and then want to immediately share an item to that group. In 99% of cases, this should only fire a single request. The collaborators pane had implemented this polling before sharing the entity with 1 or more groups, but had a bug where `num` wasn't specified on the polling request. This was problematic in the scenario where the user selected 11+ groups to share the entity with. The validation function expected 11 results, but AGO search would only return 10 (the default num of results when `num` param is omitted).